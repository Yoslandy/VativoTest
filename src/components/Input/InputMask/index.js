import React, { useState } from "react"
import { Field } from "formik"
import PropTypes from "prop-types"
import MaskedInput from "react-text-mask"

export default function InputMask(props) {
    const {
        name,
        type,
        placeholder,
        className,
        error,
        onFocus,
        mask,
        label,
        hint,
    } = props
    const [focusStyles, setFocusStyles] = useState("")

    const handleActive = () => {
        setFocusStyles("input-active")
        onFocus()
    }

    const handleBlur = () => {
        setFocusStyles("")
    }

    const getErrorStyles = () =>
        error ? "input-error" : "border-[1px] border-primary-50"

    const renderError = () =>
        error ? <span className="formik-error-message">{error}</span> : null

    return (
        <>
            <div className="flex flex-col">
                <label className="font-roboto text-[12px] text-neutral-gray-100 uppercase font-medium">
                    {label}
                </label>
                <Field name={name}>
                    {({ field }) => (
                        <MaskedInput
                            {...field}
                            mask={mask}
                            placeholder={placeholder}
                            type={type}
                            className={`input ${focusStyles} ${className} ${getErrorStyles()} mb-1`}
                            onFocus={handleActive}
                            onBlur={handleBlur}
                        />
                    )}
                </Field>
                {hint && (
                    <span className="font-roboto text-[14px] text-neutral-gray-100 text-center">
                        {hint}
                    </span>
                )}
                {renderError()}
            </div>
        </>
    )
}

InputMask.propTypes = {
    name: PropTypes.string,
    type: PropTypes.oneOf(["text", "number", "email"]),
    placeholder: PropTypes.string,
    className: PropTypes.string,
    error: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
    onFocus: PropTypes.func,
    mask: PropTypes.oneOfType([PropTypes.array, PropTypes.func]).isRequired,
    label: PropTypes.string,
    hint: PropTypes.string,
}

InputMask.defaultProps = {
    name: "name",
    type: "text",
    placeholder: "Jane",
    className: "",
    error: false,
    onFocus: () => {},
    label: "",
    hint: "",
}
