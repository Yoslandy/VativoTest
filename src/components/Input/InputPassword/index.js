import React, { useState } from "react"
import { Field, ErrorMessage } from "formik"
import PropTypes from "prop-types"
import { sizes } from "./utils"
import Icon from "../../Icon"

export default function InputPassword(props) {
    const { name, placeholder, className, error, size, onFocus, label, hint } =
        props

    const [isPasswordType, setIsPasswordType] = useState(true)

    const handleActive = () => {
        onFocus()
    }

    const getErrorStyles = () =>
        error ? "input-error" : "border-[1px] border-primary-50"

    const renderError = () =>
        error && (
            <span className="formik-error-message">
                <ErrorMessage name={name} />
            </span>
        )

    return (
        <>
            <div className="flex flex-col">
                <label className="font-roboto text-[12px] text-neutral-gray-100 uppercase font-medium">
                    {label}
                </label>
                <div
                    className={`password-wrapper ${className} ${getErrorStyles()} ${
                        sizes[size]
                    }`}
                >
                    <Field
                        name={name}
                        placeholder={placeholder}
                        type={isPasswordType ? "password" : "text"}
                        className={`password-input ${
                            size === "big" ? "h-[41px]" : "h-[38px]"
                        }`}
                        onFocus={handleActive}
                    />
                    <div className="password-icon cursor-pointer">
                        <Icon
                            type={isPasswordType ? "eye" : "eye-slash"}
                            onClick={() => setIsPasswordType(!isPasswordType)}
                        />
                    </div>
                </div>
                {hint && (
                    <span className="font-roboto text-[14px] text-neutral-gray-100 text-center">
                        {hint}
                    </span>
                )}
                {renderError()}
            </div>
        </>
    )
}

InputPassword.propTypes = {
    name: PropTypes.string,
    placeholder: PropTypes.string,
    className: PropTypes.string,
    error: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
    size: PropTypes.oneOf(["small", "big"]),
    onFocus: PropTypes.func,
    label: PropTypes.string,
    hint: PropTypes.string,
}

InputPassword.defaultProps = {
    name: "name",
    placeholder: "Jane",
    className: "",
    error: false,
    size: "big",
    onFocus: () => {},
    label: "",
    hint: "",
}
