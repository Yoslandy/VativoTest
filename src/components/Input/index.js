import React, { useState } from "react"
import { Field } from "formik"
import PropTypes from "prop-types"

export default function Input(props) {
    const {
        name,
        type,
        placeholder,
        className,
        error,
        onFocus,
        label,
        hint,
        component,
        rows,
    } = props
    const [focusStyles, setFocusStyles] = useState("")

    const handleActive = () => {
        setFocusStyles("input-active")
        onFocus()
    }

    const handleBlur = () => {
        setFocusStyles("")
    }

    const getErrorStyles = () =>
        error ? "input-error" : "border-[1px] border-primary-50"

    const renderError = () =>
        error ? <span className="formik-error-message">{error}</span> : null

    return (
        <>
            <div className="flex flex-col">
                <label className="font-roboto text-[12px] text-neutral-gray-100 uppercase font-medium">
                    {label}
                </label>
                <Field
                    name={name}
                    placeholder={placeholder}
                    type={type}
                    className={`input ${focusStyles} ${className} ${getErrorStyles()}`}
                    onFocus={handleActive}
                    onBlur={handleBlur}
                    component={component}
                    rows={rows}
                />
                {hint && (
                    <span className="font-roboto text-[14px] text-neutral-gray-100 text-center">
                        {hint}
                    </span>
                )}
                {renderError()}
            </div>
        </>
    )
}

Input.propTypes = {
    name: PropTypes.string,
    type: PropTypes.oneOf(["text", "number", "email"]),
    placeholder: PropTypes.string,
    className: PropTypes.string,
    error: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
    onFocus: PropTypes.func,
    label: PropTypes.string,
    hint: PropTypes.string,
    component: PropTypes.string,
    rows: PropTypes.number,
}

Input.defaultProps = {
    name: "name",
    type: "text",
    placeholder: "Jane",
    className: "",
    error: false,
    onFocus: () => {},
    label: "",
    hint: "",
    component: "input",
    rows: 20,
}
