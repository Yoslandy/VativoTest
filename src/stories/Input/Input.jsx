import React from "react"
import PropTypes from "prop-types"
import { Formik, Form } from "formik"
import Input from "../../components/Input"

export default function CustomInput(props) {
    const { name, ...rest } = props
    return (
        <Formik
            initialValues={{
                [name]: "",
            }}
        >
            <Form>
                <Input name={name} {...rest} />
            </Form>
        </Formik>
    )
}

CustomInput.propTypes = {
    name: PropTypes.string,
    className: PropTypes.string,
}

CustomInput.defaultProps = {
    name: "input",
    className: "",
}
