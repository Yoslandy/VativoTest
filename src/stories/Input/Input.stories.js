import React from "react"

import "@fortawesome/fontawesome-free/css/all.min.css"
import "../../assets/styles/build.css"

import Input from "./Input"

export default {
    title: "Example/Input",
    component: Input,
    argTypes: {
        name: "Input",
        placeholder: "",
        className: "",
    },
}

function Template(args) {
    return <Input {...args} />
}

export const Default = Template.bind({})
Default.args = {
    name: "Input",
    placeholder: "Placeholder",
}

export const Bordered = Template.bind({})
Bordered.args = {
    name: "Input",
    placeholder: "Placeholder",
    className: "border border-sky-600",
}
