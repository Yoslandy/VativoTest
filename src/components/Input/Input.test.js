import React from "react"
import { MemoryRouter } from "react-router-dom"
import renderer from "react-test-renderer"
import { Formik, Form } from "formik"
import Input from "./index"

test("Renders Input form correctly", () => {
    const component = renderer.create(
        <MemoryRouter>
            <Formik
                initialValues={{
                    input: "",
                }}
            >
                <Form>
                    <Input name="input" />
                </Form>
            </Formik>
        </MemoryRouter>
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})
