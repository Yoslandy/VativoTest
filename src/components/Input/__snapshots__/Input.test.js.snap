// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Input form correctly 1`] = `
<form
  action="#"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <input
    className="input   border-[1px] border-primary-50"
    name="input"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Jane"
    type="text"
    value=""
  />
</form>
`;
